src/ApacheOrcDotNet.OptimizedReader/OrcReader.cs

```diff
@@ -35,93 +40,452 @@ public OrcReader(OrcReaderConfiguration configuration, IByteRangeProvider byteRa
             if (_fileTail.Footer.Types[0].Kind != ColumnTypeKind.Struct)
                 throw new InvalidDataException($"The base type must be {nameof(ColumnTypeKind.Struct)}");
 
-            ColumnDetails = _fileTail.Footer.Types[0].FieldNames
-                .Select((name, i) =>
-                {
-                    var subType = (int)_fileTail.Footer.Types[0].SubTypes[i];
-                    var subTypeKind = _fileTail.Footer.Types[subType].Kind;
-                    return new ColumnDetail(ColumnId: subType, Name: name, ColumnType: subTypeKind);
-                })
-                .ToList();
+            _protoColumns = _fileTail.Footer.Types[0].FieldNames.Select((name, i) =>
+            {
+                var subType = (int)_fileTail.Footer.Types[0].SubTypes[i];
+                var subTypeKind = _fileTail.Footer.Types[subType].Kind;
+                return (subType, name, subTypeKind);
+            }).ToDictionary(i => i.name.ToLower(), i => i);
```

Comment:
`i` is not a good variable name in `ToDictionary`.  `fieldDetail` or `x` would be more understandable.